src/code/z_message_nes.c

```diff
@@ -1574,6 +1574,7 @@ void Message_DecodeNES(PlayState* play) {
             Message_LoadPluralRupeesNES(play, &decodedBufPos, &charTexIndex, &spA4);
         } else if (curChar == 0xDF) {
             for (i = 0; i < 5; i++) {
+                //! @bug OoB read & write for i == 4, digits array is only 4 elements
                 digits[i] = gSaveContext.save.saveInfo.bomberCode[i];
```

Comment:
In most cases, this issue was artificial and could just be fixed by fixing the stack, which we've done a few times for these decode functions now. However, `s16 digits[4]; // spC8` needs to be at stack 0xC8, and `s16 numLines; // spD0` needs to be at stack 0xD0, so there's not really any wiggle room for stack. Plus, replacing `numLines` with `digits[5]` doesn't work since `numLines` carries over in the loop.

So I think this might be an authentic OoB? I'd like a bit more time with this function and maybe other experienced decompers can take a look too for a second opinion. But interesting find. Thanks

Reply:
Yeah, I'm not seeing an easy way to fix this. This comment is fine then.

Something to take in consideration that we are trying to get everybody who has contributed / wants to contribute to MM decomp to agree to licensing the repo before merging PRs from people who has not yet agreed to it. It is better explained here: https://github.com/louist103/mm-licensing/issues/1.

Could you please read about it and post in that issue if you agree or not. Thanks!

Reply:
@engineer124 Sorry for the delay, agreed that license 👍